internal/commands/push/command_test.go

```diff
@@ -1429,6 +1431,89 @@ func TestPushCommandDeployDraftAndWait(t *testing.T) {
 	})
 }
 
+func TestPushCommandImportDependenciesAndWait(t *testing.T) {
+	groupID, appID, uploadPath := "groupID", "appID", "node_modules.zip"
+	testError := errors.New("something bad happened")
+
+	t.Run("should return an error with a client that fails to import dependencies", func(t *testing.T) {
+		realmClient := mock.RealmClient{}
+
+		var capturedGroupID, capturedAppID, capturedUploadPath string
+		realmClient.ImportDependenciesFn = func(groupID, appID, uploadPath string) error {
+			capturedGroupID = groupID
+			capturedAppID = appID
+			capturedUploadPath = uploadPath
+			return testError
+		}
+		err := importDependenciesAndWait(nil, realmClient, appRemote{groupID, appID}, uploadPath)
+		assert.Equal(t, testError, err)
+
+		t.Log("and should properly pass through the expected inputs")
+		assert.Equal(t, groupID, capturedGroupID)
+		assert.Equal(t, appID, capturedAppID)
+		assert.Equal(t, uploadPath, capturedUploadPath)
+	})
+
+	t.Run("with a client that successfully imports dependencies", func(t *testing.T) {
+		realmClient := mock.RealmClient{}
+		realmClient.ImportDependenciesFn = func(groupID, appID, uploadPath string) error {
+			return nil
+		}
+
+		t.Run("but fails to finish the installation", func(t *testing.T) {
+			realmClient.DependenciesInstallationFn = func(groupID, appID string) (realm.DependenciesInstallation, error) {
+				return realm.DependenciesInstallation{}, testError
+			}
+
+			out, ui := mock.NewUI()
+			err := importDependenciesAndWait(ui, realmClient, appRemote{groupID, appID}, uploadPath)
```

Comment:
I imagine you introduced the `importDependenciesAndWait` helper for testing purposes, but this should be something thats just as easily testable with `cmd.Handler(...)` (if we choose to get rid of the helper all together)

Reply:
I took the same approach / structure as the `deployDraftAndWait` function [here to stay consistent.](https://github.com/10gen/realm-cli/blob/master/internal/commands/push/command.go#:~:text=func-,deployDraftAndWait,-(ui%20terminal.UI). I did introduce the [waitForInstallation function for testing](https://github.com/gagik/realm-cli/commit/11961946d9ab94177a0f5c31856630764840d823#:~:text=func-,waitForInstallation,-(realmClient%20realm.Client) which I will remove since it's only being used once.

Reply:
so we have patterns in the `if cmd.inputs.IncludeHosting` and `if cmd.inputs.ResetCDNCache` blocks as well that I think equally apply, at the very least I want to make sure we print logs in a similar manner/style to those blocks

if you find this extra function dramatically helps testing then I'm totally fine leaving it as is (I'll be the first one to tell you I was a little "helper function crazy" earlier on in this codebase 😬 )

Reply:
ah okay, did not notice how `cmd.Handler(...)` works, will give this a try.